### PR TITLE
Improve Dockerfile caching

### PR DESCRIPTION
**Pull request by AurelienGasser**
Speed up builds by finding the largest common denominator between Dockerfiles.

Regarding the order of the files copied: make the order the same as with other Dockerfiles (to improve caching, though it is a very minor improvement)

### CODE REVIEW

**Comment by AurelienGasser** [comment]
Tested with end-to-end tests: :+1:

File: docker/substra-backend/Dockerfile [hunk 1]
```diff
@@ -11,11 +11,11 @@ COPY ./backend/requirements.txt /usr/src/app/.
```
Comment (inalgnu) [inline docker/substra-backend/Dockerfile:11]:
```suggestion
RUN apt-get update \
    && apt-get install -y --no-install-recommends \
        python3 \
        python3-pip  \
        python3-dev \
        build-essential \
        libssl-dev \
        libffi-dev \
       libxml2-dev \
       libxslt1-dev \
       zlib1g-dev \
       git \
       curl \
       netcat \
    && apt-get clean \
    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* /usr/share/doc/*

```
could be done in one layer + reduce layer size by cleaning noisy files

**Comment by AurelienGasser** [comment]
I did a few tests:

| Disk usage (MB) | Command | Comment |
| ------------- | ---------  | ----- |
| 932 | (startup) |   |
| 949  |  `apt-get update`  |   |
| 1052    | `apt-get install -y python3 python3-pip python3-dev build-essential libssl-dev libffi-dev libxml2-dev libxslt1-dev zlib1g-dev` | 1037 MB with `--no-install-recommend`
| 1052 | `apt-get clean`  |    |   
| 969 | `rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* /usr/share/doc/*`  |    |   


It looks like:

- `apt-get clean` has no effect on disk usage
- the final `rm -rf` is super useful

**Comment by AurelienGasser** [comment]
I tried another approach in a new PR: https://github.com/SubstraFoundation/substra-backend/pull/219

**Comment by AlexandrePicosson** [comment]
Can we close this PR as you are now working in https://github.com/SubstraFoundation/substra-backend/pull/219 ?

**Comment by AurelienGasser** [comment]
Closing in favor of #219

**Pull request closed without merging**